src/renderer/settings/control-panel-settings-component.ts

```diff
@@ -0,0 +1,71 @@
+import Vue from "vue";
+import { vueEventDispatcher } from "../vue-event-dispatcher";
+import { VueEventChannels } from "../vue-event-channels";
+import { UserConfigOptions } from "../../common/config/user-config-options";
+import { TranslationSet } from "../../common/translation/translation-set";
+import { UserConfirmationDialogParams, UserConfirmationDialogType } from "./modals/user-confirmation-dialog-params";
+import { deepCopy } from "../../common/helpers/object-helpers";
+import { defaultControlPanelOptions } from "../../common/config/control-panel-options";
+import { SettingOsSpecific } from "./settings-os-specific";
+import { platform } from "os";
+
+export const controlPanelSettingsComponent = Vue.extend({
+    data() {
+        return {
+            settingName: SettingOsSpecific.ControlPanel.replace(`${platform()}:`, ""),
+            visible: false,
+        };
+    },
+    methods: {
+        toggleEnabled() {
+            const config: UserConfigOptions = this.config;
+            config.controlPanelOptions.isEnabled = !config.controlPanelOptions.isEnabled;
+            this.updateConfig();
+        },
+        resetAll() {
+            const translations: TranslationSet = this.translations;
+            const userConfirmationDialogParams: UserConfirmationDialogParams = {
+                callback: () => {
+                    const config: UserConfigOptions = this.config;
+                    config.controlPanelOptions = deepCopy(defaultControlPanelOptions);
+                    this.updateConfig();
+                },
+                message: translations.resetPluginSettingsToDefaultWarning,
+                modalTitle: translations.resetToDefault,
+                type: UserConfirmationDialogType.Default,
+            };
+            vueEventDispatcher.$emit(VueEventChannels.settingsConfirmation, userConfirmationDialogParams);
+        },
+        updateConfig() {
+            vueEventDispatcher.$emit(VueEventChannels.configUpdated, this.config);
```

Comment:
When activating the plugin the `refreshIndex` method of the plugin should be called but this doesn't happen yet. You should add here a third parameter `true` and a fourth parameter `PluginType.ControlPanel`, like this:

``` typescript
updateConfig() {
    vueEventDispatcher.$emit(VueEventChannels.configUpdated, this.config, true, PluginType.ControlPanel);
}
```

The third parameter specifies if a index refresh is needed or not. The fourth parameter specifies the plugin which should be refreshed.

See here: https://github.com/oliverschwendener/ueli/blob/1a2878baeb0c4ffa03bf54610303c7ffaecd0eee/src/renderer/renderer.ts#L142-L150